product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2015 OdooMRP team
+# Copyright 2015 AvanzOSC
+# Copyright 2015 Tecnativa
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    @api.multi
+    def _get_price_from_supplierinfo(self, partner_id):
+        if partner_id:
+            for product in self:
+                supplierinfo = self.env['product.supplierinfo'].search(
+                    ['|', ('product_tmpl_id', '=', product.product_tmpl_id.id),
+                     ('product_id', '=', product.id),
+                     ('type', '=', 'customer'),
+                     ('name', '=', partner_id)])
+                if supplierinfo:
+                    return supplierinfo.price
```

Comment:
Ok, you want to return the first supplierinfo that you find. But what if you find more than one? I suggest adding `limit=1` to the search arguments